kuwala/core/canvas/src/pages/DataSourceConfiguration.js

```diff
@@ -1,57 +1,115 @@
+                                            <div className={'flex flex-row h-full w-full items-center space-y-8'}
+                                                key={conParams.id}
+                                            >
+                                                <div className={'w-1/6'}>
+                                                    <span className={'text-lg capitalize font-bold'}>
+                                                        {conParams.id}
+                                                    </span>
+                                                </div>
+                                                <div className={'w-5/6'}>
+                                                    <Field
+                                                        name={`connection_parameters[${i}].value`}
+                                                        type={conParams.id === 'password' ? 'password' : 'text'}
+                                                        className={'w-full px-4 py-2 border-2 border-kuwala-green text-gray-800 rounded-lg focus:outline-none'}
+                                                        placeholder={conParams.id}
+                                                        key={conParams.id}
+                                                    />
+                                                </div>
+                                            </div>
+                                        )
+                                    })}
+                                    <div className={'flex flex-row justify-between mt-6'}>
+                                        <button
+                                            className={'bg-white border-2 border-kuwala-green rounded-md px-3 py-2 mt-4 hover:text-stone-300 hover:bg-kuwala-bg-gray'}
+                                            onClick={async ()=> {
+                                                await testConfigConnection({
+                                                    dataSourceId: selectedSource.id,
+                                                    config: values.connection_parameters
+                                                })
+                                            }}
+                                            type={'button'}
+                                        >
+                                            <span className={'text-kuwala-green'}>
+                                                Test connection
+                                            </span>
+                                        </button>
+                                        <button
+                                            className={'bg-kuwala-green rounded-md px-3 py-2 mt-4'}
+                                            type={'submit'}
+                                        >
+                                            <span className={'text-white hover:text-stone-300'}>Save</span>
```

Comment:
Render a spinner while waiting for a response and go back to the overview screen.